test/models/ArrivalUpdateSpec.scala

```diff
@@ -0,0 +1,179 @@
+import org.scalacheck.Arbitrary.arbitrary
+import org.scalatest.matchers.must.Matchers
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+import play.api.libs.json.JsObject
+import play.api.libs.json.JsString
+import play.api.libs.json.Json
+
+class ArrivalUpdateSpec extends SpecBase with Matchers with ScalaCheckDrivenPropertyChecks with ModelGenerators with FreeSpecDiscipline {
+
+  val arrivalStatusUpdate: Gen[ArrivalUpdate] =
+    for {
+      arrivalUpdate <- arbitrary[ArrivalUpdate]
+      arrivalStatus <- arbitrary[ArrivalStatus]
+    } yield arrivalUpdate.copy(arrivalUpdate = Some(arrivalStatus))
+
+  val messageStatusUpdate =
+    for {
+      arrivalUpdate <- arbitrary[ArrivalUpdate]
+      messageStatus <- arbitrary[MessageStatusUpdate]
+    } yield arrivalUpdate.copy(messageUpdate = Some(messageStatus))
+
+  def noArrivalUpdate(gen: Gen[ArrivalUpdate] = arbitrary[ArrivalUpdate]): Gen[ArrivalUpdate] =
+    gen.map(_.copy(arrivalUpdate = None))
+
+  def noMessageUpdate(gen: Gen[ArrivalUpdate] = arbitrary[ArrivalUpdate]): Gen[ArrivalUpdate] =
+    gen.map(_.copy(messageUpdate = None))
+
+  implicit val eqArrivalStatusUpdate: Eq[ArrivalUpdate] =
+    new Eq[ArrivalUpdate] {
+      override def eqv(x: ArrivalUpdate, y: ArrivalUpdate): Boolean = x == y
+    }
+
+  "ArrivalUpdate" - {
+
+    "combine" - {
+      checkAll("Semigroup behaviour", SemigroupTests[ArrivalUpdate].semigroup)
+
+      "uses the arrival status on the right when both are defined" in {
+        forAll(arrivalStatusUpdate, arrivalStatusUpdate) {
+          (lhs, rhs) =>
+            lhs.combine(rhs).arrivalUpdate mustEqual rhs.arrivalUpdate
+        }
+      }
+
+      "uses the arrival status on the left when the right is not are defined" in {
+        forAll(arrivalStatusUpdate, noArrivalUpdate()) {
+          (lhs, rhs) =>
+            lhs.combine(rhs).arrivalUpdate mustEqual lhs.arrivalUpdate
+        }
+      }
+
+      "is not defined when both arrival status are not defined" in {
+        forAll(noArrivalUpdate(), noArrivalUpdate()) {
+          (lhs, rhs) =>
+            lhs.combine(rhs).arrivalUpdate must not be (defined)
+        }
+
+      }
+
+      "uses the message status on the right when both are defined" in {
+        forAll(messageStatusUpdate, messageStatusUpdate) {
+          (lhs, rhs) =>
+            lhs.combine(rhs).messageUpdate mustEqual rhs.messageUpdate
+        }
+      }
+
+      "uses the message status on the left when the right is not are defined" in {
+        forAll(messageStatusUpdate, noMessageUpdate()) {
+          (lhs, rhs) =>
+            lhs.combine(rhs).messageUpdate mustEqual lhs.messageUpdate
+        }
+      }
+
+      "is not defined when both message status are not defined" in {
+        forAll(noMessageUpdate(), noMessageUpdate()) {
+          (lhs, rhs) =>
+            lhs.combine(rhs).messageUpdate must not be (defined)
+        }
+      }
+
+      "THIS MAKES IT A MONIOD" ignore {
```

Comment:
@amaalali Is this test supposed to be ignored

Reply:
Whoops, wasn't supposed to commit that, I'll remove it 😬